protos/flyteidl/admin/execution.proto

```diff
@@ -49,6 +49,21 @@ message ExecutionRelaunchRequest {
     string name = 3;
 }
 
+// Request to recover the referenced execution.
+message ExecutionRecoverRequest {
+    // Identifier of the workflow execution to recover.
+    core.WorkflowExecutionIdentifier id = 1;
+
+    // User provided value for the recovered execution.
+    // If none is provided the system will generate a unique string.
+    // +optional
+    string name = 2;
+
+    // Which subworkflow node (if any) launched this execution
+    core.NodeExecutionIdentifier parent_node_execution = 3;
```

Comment:
can we add a comment for why this is necessary?  it's not immediately obvious.

Reply:
done, let me know if this is clearer

Reply:
Not really, but I think it's because I don't understand this recovery feature/what a parent_node_execution is.

Reply:
I agree with Yee, this is confusing! Why do we need to know the parent at all. If you just recover one of the child workflows, then it should simply recover the child workflow! I do not think we should try and link it to the parent - so node execution identifier should not be required?

Reply:
This is to mimic the existing behavior when a parent workflow launches a child one. We pass through execution metadata like the principal

Reply:
I understand why we need to overwrite this, @kumare3. When you recover Exec1 that has a node that, ran Exec2 and then we failed somewhere in there... when we recover Exec1', and we come to that node, it should launch Exec2' in recover mode... but if all it did was call the Recover endpoint on Exec2, then it'll look like it's associated with the node in Exec1 not Exec1'... 

This sounds like we want to override _somethings_ within the `ExecutionSpec` of the recovered execution... and right now we think we only need to overwrite [parent_node_execution](https://github.com/flyteorg/flyteidl/blob/f4809c1a52073ec80de41be109bccc6331cdbac3/protos/flyteidl/admin/execution.proto#L200).

But I guess we might even want to overwrite [reference_execution](https://github.com/flyteorg/flyteidl/blob/f4809c1a52073ec80de41be109bccc6331cdbac3/protos/flyteidl/admin/execution.proto#L204) later on... so maybe we should have a way to overwrite all `ExecutionMetadata` fields?

Reply:
fair point @EngHabu do you mean we should pass an Execution Metadata instead here? I do question whether we want to overwrite the reference execution, admin can implicitly do this already based on it being the RecoverExecution endpoint, it seems error-prone to leave it to the caller to set that themselves

Reply:
I did mean that yes..
What should the reference_execution be? should it be the execution we are recovering from? or the "parent" execution that is sending this recover request?

Reply:
in this case the parent. the execution that we're recovering from is the source execution id (stored separately)